Makefile

```diff
@@ -1,29 +1,47 @@
-GOPATH=$(shell pwd)/vendor:$(shell pwd)
 GOBIN=$(shell pwd)/bin
+GOPATH := $(shell go env GOPATH)
+GOOS ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
 GOFILES=$(wildcard *.go)
 GONAME=dex-k8s-authenticator
+IMAGE_NAME=mesosphere/dex-k8s-authenticator
 TAG=latest
 
-all: build 
+export GO111MODULE ?= on
+export GOPRIVATE ?= github.com/mesosphere
+
+KONVOY_ASYNC_AUTH_VERSION ?= v0.1.3
+
+all: build
 
 get:
-	@GOPATH=$(GOPATH) GOBIN=$(GOBIN) go get -d .
+	@go get -d .
+
+konvoy-async-auth:
+	@rm -rf _build/konvoy-async-auth*
+	@gh release download $(KONVOY_ASYNC_AUTH_VERSION) -R https://github.com/mesosphere/konvoy-async-auth -D _build/
```

Comment:
This expecting `gh` binary to be on a local system? If so can we have a target to download it locally?